### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by daniel-rikowski** [Shopify/bootsnap#52]
In the light of #50 I wonder if the missing support for Windows is a final decision or if you would **accept pull requests for Windows support?** (At least partial support, see below)  

I'm asking because I could get the path pre-scanning feature working on Windows with [Ruby 2.4.1 (using MSYS2)](https://github.com/oneclick/rubyinstaller2) in about 30 minutes and some minor code changes. (Mostly making the C code compile and fixing two root path detection lines in Ruby)

The speed improvements were dreamlike!  On a fast Windows workstation I could bring the startup time of my Rails app down to 10 seconds from 30 seconds and on a very slow laptop down to 25 seconds from originally 2 minutes. (🐌 -> 🚀)

The compilation caching did create the cache files, but reading them results in a "bad file descriptor" error in the C code, which I couldn't track down (yet). But still, the speed improvements of the path scanning alone warrant the use of bootsnap on Windows platforms. (IMO)

The only downside is detecting an absolute path: Under a Unix-like platform it's just checking for a leading slash, but on Windows it's a little more involved. Currently I'm checking `path[1..2] == ':/'` but that might not be sufficient and - perhaps more importantly - slows down the check on all platforms. 

In any case: Thank you for this awesome gem!

**Comment by joostliketoast** [comment]
not directly relevant to this gem. But a good solution could be to run WSL which is all sorts of amazing.
it runs my rails productions environments without a hitch( File I/O is a bit slow at the moment but the next release should solve that)

**Comment by jules2689** [comment]
We would entertain the possibility of windows support :) PRs are welcome and we can discuss the best course of action.

It is absolutely crucial that it doesn't impact unix platforms like you mention - but we could perhaps use a separate C file for windows? I'm not sure how gems normally handle windows.

**Comment by daniel-rikowski** [comment]
The recent changes to bootsnap made it a lot easier to maintain Windows compatibility. See #60 
With this I could use all bootsnap features, even the compilation caching.

Here's what I did:

On the Ruby side only one place needed a change to the absolute path detection. I implemented it by checking for a colon at the second character. This is not perfect, but someone using UNC paths most likely isn't concerned with speed. Unfortunately I had to add a separate method call, but the performance hit should be negligible. (🤞)

On the C side I had to mark all open file descriptors as binary and replace a mkdir call which has no mode parameter on Windows. I also return 0 for the OS version hash. I'm not sure *what* to return here, I guess it could be some internal MSYS version marker.
